java/src/jmri/jmrix/loconet/LnSensor.java

```diff
@@ -68,17 +68,31 @@ public void requestUpdateFromLayout() {
      */
     @Override
     public void setKnownState(int s) throws jmri.JmriException {
-        // send OPC_INPUT_REP with new state to this address
-        LocoNetMessage l = new LocoNetMessage(4);
-        l.setOpCode(LnConstants.OPC_INPUT_REP);
-        a.insertAddress(l);
-        // set state
-        if ((s == Sensor.ACTIVE) ^ _inverted) {
-            l.setElement(2, l.getElement(2) | 0x10);
-        } // otherwise is already OK
-        l.setElement(2, l.getElement(2) | 0x40);
-        // send
-        tc.sendLocoNetMessage(l);
+        switch (s) {
+            case Sensor.ACTIVE:
+                if (getRawState() != Sensor.ACTIVE) {
+                      setOwnState(Sensor.ACTIVE);
+                }
+                break;
+            case Sensor.INACTIVE:
+                if (getRawState() != Sensor.INACTIVE) {
+                    setOwnState(Sensor.INACTIVE);
+                }
+                break;
+            case Sensor.UNKNOWN:
+                if (getRawState() != Sensor.UNKNOWN) {
+                    setOwnState(Sensor.UNKNOWN);
+                }
+                break;
+            case Sensor.INCONSISTENT:
+                if (getRawState() != Sensor.INCONSISTENT) {
+                    setOwnState(Sensor.INCONSISTENT);
+                }
+                break;
+            default:
+                log.error("LnSensor given invalid state[{}] setting to INCONSISTENT", s);
+                setOwnState(Sensor.INCONSISTENT);
+        }
     }
```

Comment:
Removal of `tc.sendLocoNetMessage(l)` is a significant change.  It seems inappropriate to me.  

Without sending a message, other LocoNet agents are not aware of the change sensor state, giving an inconsistent view of the sensor's state across different LocoNet devices.  This is a significant change to JMRI behavior!

Reply:
reverted. I didnt understnad the significance of other devices, thought it was other internal listeners.